dotnet-efcore/RecordsAsModelClasses/RecordsAsModelClasses/Controllers/v1/CarsController.cs

```diff
@@ -0,0 +1,101 @@
+﻿using AutoMapper;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using RecordsAsModelClasses.Core.Context;
+using RecordsAsModelClasses.Core.DTOs;
+using RecordsAsModelClasses.Core.Entities.Records;
+
+namespace RecordsAsModelClasses.Controllers.v1;
+
+[Route("api/v1/cars")]
+[ApiController]
+public class CarsController : ControllerBase
+{
+    private readonly CarDbContext _context;
+    private readonly IMapper _mapper;
+
+    public CarsController(CarDbContext context, IMapper mapper)
+    {
+        _context = context;
+        _mapper = mapper;
+    }
+
+    [HttpPost]
+    public async Task<IActionResult> CreateCarAsync([FromBody] UpsertCarDto carDto)
+    {
+        var car = _mapper.Map<UpsertCarDto, Car>(carDto);
```

Comment:
```suggestion
        var car = new Car(carDto.Make, carDto.Model, carDto.Year);
```

Reply:
Hello @karimkod, I've made the changes as suggested.
Please review.
Thanks!